index.js

```diff
@@ -348,15 +371,25 @@ function opts (pragma) {
           events: {}
         };
         let props = opt[key];
-        let lambdaKeys = Object.keys(props);
-        lambdaKeys.forEach(eventName => {
-          let filterPairs = props[eventName];
-          lambda.events[eventName] = [];
-          for (let i = 0; i < filterPairs.length - 1; i++) {
-            if (i % 2 === 1) continue;
-            lambda.events[eventName].push([ filterPairs[i], filterPairs[i + 1] ]);
-          }
-        });
+        // if filter options for an event exist, the props are returned as an Array instead of an Object and need to be handled differently
+        if (Array.isArray(props)) {
+          let lambdaKeys = props;
+          lambdaKeys.forEach(eventName => {
+            lambda.events[eventName] = [];
+            obj.lambdas.push(lambda);
+          });
+        } else {
+          let lambdaKeys = Object.keys(props);
```

Comment:
Do you think we can keep the original code (instead of the if/else) and do something like this instead?
```suggestion
          let lambdaKeys = Array.isArray(props) ? props : Object.keys(props);
```

That'd make us not repeat any logic in two different branches.

Reply:
You also have to iterate through each filter pair and add them to the lambda object when visiting the Object.keys `else` block and i'm not sure how to cleanly do this without a much larger refactor.

Reply:
You can do the `filterPairs` thing (and the for) only if it's not an array.